packages/deployer/src/compiler.ts

```diff
@@ -162,49 +171,69 @@ export class Compiler {
         }
         const solcInstance = solc.setupMethods(requireFromString(solcjs, compilerBinFilename));
 
-        logUtils.log(`Compiling ${fileName} with Solidity v${solcVersion}...`);
-        const source = this._contractSources[fileName];
-        const input = {
-            [fileName]: source,
-        };
-        const sourcesToCompile = {
-            sources: input,
+        logUtils.log(`Compiling ${contractName} with Solidity v${solcVersion}...`);
+        const source = this._contractSources[contractName].source;
```

Comment:
```
const contractSource = this._contractSources[contractName];
const source = contractSource.source;
const absoluteFilePath = contractSource.absoluteFilePath;
```

Reply:
Are we guaranteed there is a corresponding entry in  `this._contractSources`?

Reply:
Yes